src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -15001,35 +15001,204 @@ def stack(
                 "Snowpark pandas doesn't support multiindex columns in stack API"
             )
 
-        index_names = ["index"]
+        index_names = self.get_index_names()
```

Comment:
This code change is very similar to unstack with the non multiindex case and fixes a bug with setting index name. There is an associated test added as well